src/Components/Popover.tsx

```diff
@@ -0,0 +1,104 @@
+import React, { cloneElement } from "react"
+import styled, { StyledFunction } from "styled-components"
+
+import * as fonts from "../Assets/Fonts"
+
+interface PopoverArrowPositionProps {
+  arrowOffsetTop?: string
+  arrowOffsetLeft?: string
+}
+
+interface PopoverProps
+  extends React.HTMLProps<HTMLDivElement>,
+    PopoverArrowPositionProps {
+  placement?: "top" | "right" | "bottom" | "left"
+}
+
+export class Popover extends React.Component<PopoverProps> {
+  private readonly Arrows = {
+    top: <PopoverTopArrow />,
+    right: <PopoverRightArrow />,
+    bottom: <PopoverBottomArrow />,
+    left: <PopoverLeftArrow />,
+  }
+
+  render() {
+    const {
+      placement,
+      arrowOffsetLeft,
+      arrowOffsetTop,
+      ref,
+      ...props
+    } = this.props
+
+    return (
+      <PopoverContainer {...props}>
+        {cloneElement(this.Arrows[placement], {
+          arrowOffsetLeft,
+          arrowOffsetTop,
+        })}
+
+        <PopoverInner>{this.props.children}</PopoverInner>
+      </PopoverContainer>
+    )
+  }
+}
+
+const PopoverContainer = styled.div`
+  position: relative;
+  padding: 5px;
+  display: inline-block;
+`
+
+const PopoverArrowFactory: StyledFunction<
+  PopoverArrowPositionProps & React.HTMLProps<HTMLDivElement>
+> =
+  styled.div
+
+const PopoverArrow = PopoverArrowFactory`
```

Comment:
FYI, another way to add props to a styled component without declaring an interface.
Found it [here](https://github.com/styled-components/styled-components/issues/630#issuecomment-375013384)

You can do this 
```typescript
styled.div.attrs<PopoverArrowPositionProps>({
  arrowOffsetTop: string
})`
  position: absolute;
  ...
`
```

Reply:
Interesting, I'll give it a try.